src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "../libraries/BN254.sol";
+
+
+contract IndexRegistry is IIndexRegistry {
+
+    IRegistryCoordinator public registryCoordinator;
+
+    // list of all unique registered operators
+    bytes32[] public globalOperatorList;
+
+    // mapping of quorumNumber => list of operators registered for that quorum
+    mapping(uint8 => bytes32[]) public quorumToOperatorList;
+    // mapping of operatorId => quorumNumber => index history of that operator
+    mapping(bytes32 => mapping(uint8 => OperatorIndex[])) public operatorIdToIndexHistory;
+    // mapping of quorumNumber => history of numbers of unique registered operators
+    mapping(uint8 => OperatorIndex[]) public totalOperatorsHistory;
+
+    modifier onlyRegistryCoordinator() {
+        require(msg.sender == address(registryCoordinator), "IndexRegistry.onlyRegistryCoordinator: caller is not the registry coordinator");
+        _;
+    }
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator
+    ){
+        registryCoordinator = _registryCoordinator;
+    }
+
+    /**
+     * @notice Registers the operator with the specified `operatorId` for the quorums specified by `quorumBitmap`.
+     * @param operatorId is the id of the operator that is being registered
+     * @param quorumNumbers is the quorum numbers the operator is registered for
+     * @dev access restricted to the RegistryCoordinator
+     */
+    function registerOperator(bytes32 operatorId, uint8[] memory quorumNumbers) external onlyRegistryCoordinator {
```

Comment:
Can you add the preconditions that you think are assuming here? example: https://github.com/Layr-Labs/eigenlayer-contracts/blob/d69f30ec857d525b8452d85c6549eee6057bad22/src/contracts/interfaces/IStakeRegistry.sol#L30

Reply:
done